proposed/2021/TargetFrameworkAndPlatformsFilters.md

```diff
@@ -0,0 +1,196 @@
+When browsing for packages on NuGet.org, a developer will be able to filter packages based on compatibility filters. These filters will include:
+
+1. The **generation of .NET** such as .NET Framework, .NET Core, .NET (5, 6, and later), and Any.
+2. The **supported platforms** for the generation such as .NET Standard, Windows, macOS, Linux, Android, iOS, and more.
+3. The **version number** of the generation being used such as .NET 5, .NET Core 3.1, .NET Framework 4.8, and more.
+
+This proposal only accounts for the **majority of packages on NuGet.org**. There are many unique TFMs in the .NET ecosystem, but for the sake of creating a concept that will scale with the future of .NET, many unsupported & low usage TFMs are not considered. Thus this proposal only focuses on the [most common target frameworks](https://docs.microsoft.com/dotnet/standard/frameworks#latest-versions) & omits considering TFMs with less than 5,000 packages.
+
+#### Generations of .NET
+
+There are many generations of .NET. For the sake of this proposal, we will focus on defining the three major generations:
+
+- .NET Framework:
+  - `net, net11, net20, net35, net40, net403, net45, net451, net452, net46, net461, net462, net463, net47, net471, net472, net48`
+- .NET Core:
+  - `netcoreapp1.0, netcoreapp1.1, netcoreapp2.0, netcoreapp2.1, netcoreapp2.2, netcoreapp3.0, netcoreapp3.1`
+- .NET:
+  - `net5.0, net6.0`
+- Any:
+  - .NET Framework, .NET Core, .NET
+
+![](meta/resources/TargetFrameworkAndPlatformFilters/GenerationsOfNET.png)
+
+#### Supported Platforms
+
+With .NET Core introducing support for cross-platform scenarios, we will define the following as supported platforms:
+
+- .NET Standard:
+  - `netstandard, netstandard1.0, netstandard1.1, netstandard1.2, netstandard1.3, netstandard1.4, netstandard1.5, netstandard1.6, netstandard2.0, netstandard2.1`
+- Windows:
+  - .NET Framework
+  - .NET Core
+  - .NET 5+
+    - `net5.0-windows, net6.0-windows`
+  - .NET Standard
+- macOS:
+  - .NET Core
+  - .NET 5+
+    - `net6.0-macos`
+  - .NET Standard
+- Linux:
+  - .NET Core
+  - .NET 5+
+  - .NET Standard
+- Android:
+  - .NET 6+
+    - `net6.0-android`
+  - .NET Standard
+- iOS:
+  - .NET 6+
+    - `net6.0-ios`
+  - .NET Standard
+- macCatalyst:
+  - .NET 6+
+    - `net6.0-maccatalyst`
+  - .NET Standard
+- tvOS:
+  - .NET 6+
+    - `net6.0-tvos`
+  - .NET Standard
+- Tizen:
+  - .NET 6+
+    - `net6.0-tizen`
+  - .NET Standard
+
+**Note:** .NET Standard is considered a special case that supports majority of these platforms depending on the [version of .NET standard](https://docs.microsoft.com/dotnet/standard/net-standard#net-implementation-support). If the framework precedence of a target framework supports .NET Standard, it's respective versions supported will be implicitly included.
+
+![](meta/resources/TargetFrameworkAndPlatformFilters/SupportedPlatforms.png)
```

Comment:
I can add some more words around the tabs & pill controls being used for these concepts as shown in the mockups.

Reply:
This is more about the naming convention, but `generations` as I understand is usually referred to the different versions of the frameworks. 

Things like `.NET`, `.NET Core`, `.NET Framework` are really just called frameworks. 

Is this terminology used anywhere else?

I just want to make sure we're consistent.

Reply:
Yeah we can use `Framework`, but I used generations in this spec to avoid confusion around `.NET Framework`. Perhaps we define a generation as a `framework` and a TFM as a `target framework`. :)

Reply:
Here is the official terminology. `Framework` seems a bit more accurate, but they are quite close:

> implementation for .NET “flavors” (.NET
Framework, .NET Core, Mono). This describes
the runtime, tools, libraries/frameworks for
building different groups of applications.
There are shared components across
implementations (i.e. compilers, services, .NET
Standard).

> framework for APIs (.NET Framework base
class library, .NET Core framework, Mono
base class library). Libraries is OK here too
depending on context. 

https://github.com/dotnet/brand/blob/main/dotnet-brand-guidelines.pdf